foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/sessions/UnitOfWorkImpl.java

```diff
@@ -2996,7 +2996,7 @@ public Object internalRegisterObject(Object object, ClassDescriptor descriptor,
             return null;
         }
         if (descriptor.isDescriptorTypeAggregate()) {
-            throw ValidationException.cannotRegisterAggregateObjectInUnitOfWork(object.getClass());
+           return null;
```

Comment:
While this is _probably fine_ since it already can return null, I'd prefer it if we simply caught and handled the exception in the 4 places it is used (once in the `registerObject` method, and 3 times in `MergeManager`).
That exception is thrown in a number of other places as well and I'd rather go for the slightly cleaner solution of handling it rather than simply returning null.

Reply:
Or only handle on the specific failing code paths if you know them.